requirements.txt

```diff
@@ -1,8 +1,7 @@
 vtk==9.0.1
```

Comment:
I **do** have vtk 9.0.1:
```
Requirement already satisfied: vtk in /opt/local/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (9.0.1)
```
but yet the `vtkGraphLayout.so` shared object is not provided.

It seems that you have it on your system but from elsewhere.

Reply:
@ppebay please check my last comment [here](https://github.com/DARMA-tasking/LB-analysis-framework/issues/166#issuecomment-1038894737)

I tested that as followed (in order to eliminate possibility of finding packages elsewhere):
- I did everything with just terminal 
- I created a new directory
- I git clone LBAF to this new dir
- I changed branch to `166-VTK-does-not-appear-to-provide-vtkGraphLayout`
- I created a new isolated virtual environment, then activated it
- then I installed `pip install -r requirements.txt`
- then I just ran `python src/Applications/MoveCountsViewer.py` and it worked

I was confused and asked my colleague, which I described in the comment posted above.

Reply:
Also I would suggest using venv or any other virtual environment. What that does is isolating packages for each project.
Here is the link [venv](https://docs.python.org/3/library/venv.html)

Reply:
Alright thanks for the steps, am going to replicate those.

Reply:
> Alright thanks for the steps, am going to replicate those.

Thanks, please let me know how it went

Reply:
@marcinwrobel1986 can you please check in your system where `vtkGraphLayout.so` is located? Do you have it. This is important please let me know.

Reply:
@marcinwrobel1986 I replicated all steps above and am getting the same error:
```
 python MoveCountsViewer.py -p 8 -f ../../data/stats/lb_iter -s out
[MoveCountsViewer] ### Started with Python 3.8.7
[MoveCountsViewer] Parsing command line arguments
Traceback (most recent call last):
  File "MoveCountsViewer.py", line 436, in <module>
    params.computeMoveCountsViewer()
  File "MoveCountsViewer.py", line 246, in computeMoveCountsViewer
    layout_vertices = vtk.vtkGraphLayout()
AttributeError: module 'vtkmodules.all' has no attribute 'vtkGraphLayout'
```
We need to understand where on your system is `vtkGraphLayout.so` provided. Unless we address this question we cannot make any progress here.

Reply:
Specifically once you locate where that shared object is, you can run the following:
```
 nm -A /usr/local/lib/python3.8/site-packages/vtkmodules/vtkInfovisLayout.so | grep GraphLayout
```
to verify that it's providing the requested symbol (`vtkGraphLayout`).

Reply:
> Specifically once you locate where that shared object is, you can run the following:
> 
> ```
>  nm -A /usr/local/lib/python3.8/site-packages/vtkmodules/vtkInfovisLayout.so | grep GraphLayout
> ```
> 
> to verify that it's providing the requested symbol (`vtkGraphLayout`).

I ran `nm -A libvtkInfovisLayout-9.0.so | grep GraphLayout` inside following directory: `<project-path>/LB-analysis-framework/venv/lib/python3.8/site-packages/vtkmodules`

Got the following result:
```shell
libvtkInfovisLayout-9.0.so:0000000000024440 t _GLOBAL__sub_I_vtkGraphLayout.cxx
libvtkInfovisLayout-9.0.so:00000000000244c0 t _GLOBAL__sub_I_vtkGraphLayoutStrategy.cxx
libvtkInfovisLayout-9.0.so:0000000000045a00 T _ZN14vtkGraphLayout11RequestDataEP14vtkInformationPP20vtkInformationVectorS3_
libvtkInfovisLayout-9.0.so:00000000000230fc t _ZN14vtkGraphLayout11RequestDataEP14vtkInformationPP20vtkInformationVectorS3_.cold.15
libvtkInfovisLayout-9.0.so:0000000000046520 t _ZN14vtkGraphLayout12GetTransformEv
libvtkInfovisLayout-9.0.so:0000000000045650 T _ZN14vtkGraphLayout12SetTransformEP20vtkAbstractTransform
libvtkInfovisLayout-9.0.so:0000000000046690 t _ZN14vtkGraphLayout14UseTransformOnEv
libvtkInfovisLayout-9.0.so:0000000000046560 t _ZN14vtkGraphLayout15GetUseTransformEv
libvtkInfovisLayout-9.0.so:0000000000046530 t _ZN14vtkGraphLayout15SetUseTransformEb
libvtkInfovisLayout-9.0.so:00000000000466d0 t _ZN14vtkGraphLayout15UseTransformOffEv
libvtkInfovisLayout-9.0.so:0000000000045900 T _ZN14vtkGraphLayout16IsLayoutCompleteEv
libvtkInfovisLayout-9.0.so:00000000000230ec t _ZN14vtkGraphLayout16IsLayoutCompleteEv.cold.14
libvtkInfovisLayout-9.0.so:00000000000464d0 t _ZN14vtkGraphLayout17GetLayoutStrategyEv
libvtkInfovisLayout-9.0.so:00000000000457b0 T _ZN14vtkGraphLayout17SetLayoutStrategyEP22vtkGraphLayoutStrategy
libvtkInfovisLayout-9.0.so:00000000000465f0 t _ZN14vtkGraphLayout30GetNumberOfGenerationsFromBaseEPKc
libvtkInfovisLayout-9.0.so:0000000000046570 t _ZN14vtkGraphLayout3IsAEPKc
libvtkInfovisLayout-9.0.so:0000000000045770 T _ZN14vtkGraphLayout3NewEv
libvtkInfovisLayout-9.0.so:00000000000230dc t _ZN14vtkGraphLayout3NewEv.cold.13
libvtkInfovisLayout-9.0.so:00000000000458c0 T _ZN14vtkGraphLayout8GetMTimeEv
libvtkInfovisLayout-9.0.so:00000000000464e0 t _ZN14vtkGraphLayout9GetZRangeEv
libvtkInfovisLayout-9.0.so:0000000000045fa0 T _ZN14vtkGraphLayout9PrintSelfERSo9vtkIndent
libvtkInfovisLayout-9.0.so:00000000000464f0 t _ZN14vtkGraphLayout9SetZRangeEd
libvtkInfovisLayout-9.0.so:00000000000456b0 T _ZN14vtkGraphLayoutC1Ev
libvtkInfovisLayout-9.0.so:00000000000456b0 T _ZN14vtkGraphLayoutC2Ev
libvtkInfovisLayout-9.0.so:00000000000230c0 t _ZN14vtkGraphLayoutC2Ev.cold.12
libvtkInfovisLayout-9.0.so:0000000000045630 T _ZN14vtkGraphLayoutD0Ev
libvtkInfovisLayout-9.0.so:00000000000455b0 T _ZN14vtkGraphLayoutD1Ev
libvtkInfovisLayout-9.0.so:00000000000455b0 T _ZN14vtkGraphLayoutD2Ev
libvtkInfovisLayout-9.0.so:0000000000029fb0 t _ZN22vtkGraphLayoutStrategy10InitializeEv
libvtkInfovisLayout-9.0.so:0000000000029fd0 t _ZN22vtkGraphLayoutStrategy14GetWeightEdgesEv
libvtkInfovisLayout-9.0.so:0000000000046720 T _ZN22vtkGraphLayoutStrategy14SetWeightEdgesEb
libvtkInfovisLayout-9.0.so:0000000000029fc0 t _ZN22vtkGraphLayoutStrategy16IsLayoutCompleteEv
libvtkInfovisLayout-9.0.so:0000000000029fe0 t _ZN22vtkGraphLayoutStrategy18GetEdgeWeightFieldEv
libvtkInfovisLayout-9.0.so:00000000000467f0 T _ZN22vtkGraphLayoutStrategy18SetEdgeWeightFieldEPKc
libvtkInfovisLayout-9.0.so:0000000000046bf0 t _ZN22vtkGraphLayoutStrategy30GetNumberOfGenerationsFromBaseEPKc
libvtkInfovisLayout-9.0.so:0000000000046c50 t _ZN22vtkGraphLayoutStrategy3IsAEPKc
libvtkInfovisLayout-9.0.so:0000000000046770 T _ZN22vtkGraphLayoutStrategy8SetGraphEP8vtkGraph
libvtkInfovisLayout-9.0.so:0000000000046930 T _ZN22vtkGraphLayoutStrategy9PrintSelfERSo9vtkIndent
libvtkInfovisLayout-9.0.so:00000000000468b0 T _ZN22vtkGraphLayoutStrategyC1Ev
libvtkInfovisLayout-9.0.so:00000000000468b0 T _ZN22vtkGraphLayoutStrategyC2Ev
libvtkInfovisLayout-9.0.so:0000000000046910 T _ZN22vtkGraphLayoutStrategyD0Ev
libvtkInfovisLayout-9.0.so:00000000000468e0 T _ZN22vtkGraphLayoutStrategyD1Ev
libvtkInfovisLayout-9.0.so:00000000000468e0 T _ZN22vtkGraphLayoutStrategyD2Ev
libvtkInfovisLayout-9.0.so:0000000000046710 t _ZNK14vtkGraphLayout19NewInstanceInternalEv
libvtkInfovisLayout-9.0.so:00000000000464c0 t _ZNK14vtkGraphLayout20GetClassNameInternalEv
libvtkInfovisLayout-9.0.so:0000000000046be0 t _ZNK22vtkGraphLayoutStrategy19NewInstanceInternalEv
libvtkInfovisLayout-9.0.so:0000000000046bd0 t _ZNK22vtkGraphLayoutStrategy20GetClassNameInternalEv
libvtkInfovisLayout-9.0.so:00000000002713f0 V _ZTI14vtkGraphLayout
libvtkInfovisLayout-9.0.so:00000000002716d0 V _ZTI22vtkGraphLayoutStrategy
libvtkInfovisLayout-9.0.so:000000000005f2f0 V _ZTS14vtkGraphLayout
libvtkInfovisLayout-9.0.so:000000000005f310 V _ZTS22vtkGraphLayoutStrategy
libvtkInfovisLayout-9.0.so:0000000000271408 V _ZTV14vtkGraphLayout
libvtkInfovisLayout-9.0.so:00000000002716e8 V _ZTV22vtkGraphLayoutStrategy
```

Reply:
Below a list of files and dirs in my `vtkmodules` dir
[Uploading ls_vtk_modules.txt…]()

Reply:
Thanks, most useful. 

After a complete erasure of all my PyPI packages, and a re-install of `requirements.txt`, I confirm that I have the following:
```
 nm -A /opt/local/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vtkmodules/vtkInfovisLayout.cpython-38-darwin.so | grep vtkGraphLayout
```
we find in particular 
```
/opt/local/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vtkmodules/vtkInfovisLayout.cpython-38-darwin.so: 000000000001daa0 T _PyVTKAddFile_vtkGraphLayout
```

Reply:
Hypothesis: I had an older version of `cpython` before the reinstallation.
Shouldn't the desired version of it be made part of the requirements?

Reply:
That would be equal to Python version, it shouldn't matter